### ISSUE TITLE
Handling of signatures (contact info at bottom of message)

### ISSUE THREAD

**Issue by J-J-Chiarella** [k9mail/k-9#1928]
This is not a major deal. Just a way to cut down on messy text blobs if people use the default settings when making their own signatures.

```
This is how K-9 does signatures.
-- 
signature
```

```
This is how signatures are usually done.

-- 
signature
```

I manually add the "-- " in my signatures and pre-pend the line break so my signature in the Sending options look right. "<CR><LF>--<SP><CR><LF>signature" If every other client and webmail is adding the line break, K-9 should, too.

```
Dear Person:

Hello. This email has multiple paragraphs.

Therefore, having the signature (by default) come immediately after is a bit messy-looking.

This is especially true if the signature is long.

Thank you.
-- 
Jonathan
jonathan@mail.net
Mobile: +1-500-555-6981

Senior Nitpicker
Room 45
Dept. of Scrutiny
Compulsive Industries

4103 Reice Avenue
Milltown, SX 452709-6395
```

c.f.

```
Dear Person:

Hello. This email has multiple paragraphs.

Therefore, having the signature have a line break is cleaner-looking.

This is especially true if the signature is long.

Thank you.

-- 
Jonathan
jonathan@mail.net
Mobile: +1-500-555-6981

Senior Nitpicker
Room 45
Dept. of Scrutiny
Compulsive Industries

4103 Reice Avenue
Milltown, SX 452709-6395
```

Some people purposely add several <CR><LF> in their signatures to visually separate sections of the signature and to sharply visually separate the signature as a whole from the message content. Others use desktop clients that add liberal (i.e., probably too much) white space.

It probably doesn't bode well for K-9 adoption if people complain about K-9-using correspondents' signatures being stuck to the message text and looking cluttered.

If everyone else is going against IETF RFC's and K-9 is right, then so be it. But it just seems like K-9 Mail is the odd one out from the prevailing convention.

**Comment by philipwhiuk** [comment]
I think we should definitely detect the alternative signature style if it's common. Whether we should use the alternative style is harder to answer (I have no idea if there's an RFC and I really don't care too much if there is - de facto is more important here I think).

**Comment by mvhulten** [comment]
I was surprised after configuring a signature in K-9 Mail today that my e-mail was sent without "-- \n".  In my opinion a mail program should add "-- \n" before the signature (maybe prefixed with an extra new line), because it is, and has been, a long-standing convention.  I think that [RFC 3676](https://tools.ietf.org/html/rfc3676) is relevant here.  There are several other guidelines in there that could improve K-9 Mail as well.

Many RFCs are de facto standards (even if it did not go through the whole IETF standardisation).  Does the "-- \n" convention have any significant disadvantage?  Is there a good reason to not implement this?

**Comment by Philippe-Choquette** [comment]
Agreed. First time I use an email client that doesn't use this format automatically.
